contracts/Autopay.sol

```diff
@@ -100,22 +107,29 @@ contract Autopay is UsingTellor {
      * @param _queryId id of reported data
      * @param _timestamps[] batch of timestamps array of reported data eligible for reward
      */
-    function claimOneTimeTip(bytes32 _queryId, uint256[] calldata _timestamps)external{
+    function claimOneTimeTip(bytes32 _queryId, uint256[] calldata _timestamps) external{
         require(tips[_queryId].length > 0,"no tips submitted for this queryId");
-        uint256 _reward;
         uint256 _cumulativeReward;
+        uint256 _thisReward;
+        uint256 _extraReward; // tip above stakeAmount cap
+        uint256 _stakeAmount = tellor.stakeAmount();
```

Comment:
So just a question, shouldn't it be stakeAmount + gas costs.  If gas is greater than stake amount, this solution prevents any way to tip for it right?

Reply:
yes, if gas cost (gas_report + gas_tip_claim) is above stake amount, there's no way to collect a profitable tip

Reply:
Maybe just add in tx.gasPrice * the amount of gas the txn takes?

Reply:
That would be in submitValue, right? Calc the gas used to report, convert to USD using base token SpotPrice, convert to TRB amount using TRB SpotPrice, save gasCostInTRB in flex. Then autopay can retrieve this amount from flex when tips are claimed.